The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20514209248/job/58939211412) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/58939211412) [(plain)](https://github.com/rust-lang/rust/commit/a2c9de415161c9ac855ccb0f12744f8c7f8c1a2a/checks/58939211412/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [codegen] tests/codegen-llvm/intrinsics/likely_const_assert.rs stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs:13:11: error: CHECK: expected string not found in input
// CHECK: br i1 %x, label %bb2, label %bb1, !prof ![[NUM:[0-9]+]]
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll:10:26: note: scanning from here
define void @test_assert(i1 noundef zeroext %x) unnamed_addr #0 {
                         ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll:12:2: note: possible intended match here
 br i1 %x, label %bb2, label %bb1
 ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll
Check file: /checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'likely_const_assert.acc64894c2abf645-cgu.0' 
            2: source_filename = "likely_const_assert.acc64894c2abf645-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
            4: target triple = "aarch64-unknown-linux-gnu" 
            5:  
            6: @alloc_229a7d18cf7072cde189ca7a0a5192e3 = private unnamed_addr constant [63 x i8] c"/checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs\00", align 1 
            7: @alloc_a1222f3d3751f79e8911dd06629a5c55 = private unnamed_addr constant <{ ptr, [16 x i8] }> <{ ptr @alloc_229a7d18cf7072cde189ca7a0a5192e3, [16 x i8] c">\00\00\00\00\00\00\00\05\00\00\00\07\00\00\00" }>, align 8 
            8:  
            9: ; Function Attrs: uwtable 
           10: define void @test_assert(i1 noundef zeroext %x) unnamed_addr #0 { 
check:13'0                              X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
           11: start: 
check:13'0     ~~~~~~~
           12:  br i1 %x, label %bb2, label %bb1 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:13'1      ?                                 possible intended match
           13:  
check:13'0     ~
           14: bb1: ; preds = %start 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~
           15: ; call likely_const_assert::test_assert::do_panic::runtime 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           16:  tail call fastcc void @_RNvNvNvCsePFQ7nKV49d_19likely_const_assert11test_assert8do_panic7runtime() #3 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           17:  unreachable 
check:13'0     ~~~~~~~~~~~~~
           18:  
check:13'0     ~
           19: bb2: ; preds = %start 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~
           20:  ret void 
check:13'0     ~~~~~~~~~~
           21: } 
check:13'0     ~~
           22:  
check:13'0     ~
           23: ; likely_const_assert::test_assert::do_panic::runtime 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           24: ; Function Attrs: cold noreturn uwtable 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           25: define internal fastcc void @_RNvNvNvCsePFQ7nKV49d_19likely_const_assert11test_assert8do_panic7runtime() unnamed_addr #1 { 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           26: start: 
check:13'0     ~~~~~~~
           27: ; call core::panicking::panic_fmt 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           28:  tail call void @_RNvNtCsj5hlU51TwZT_4core9panicking9panic_fmt(ptr noundef nonnull inttoptr (i64 1 to ptr), ptr noundef nonnull inttoptr (i64 1 to ptr), ptr noalias noundef nonnull readonly align 8 dereferenceable(24) @alloc_a1222f3d3751f79e8911dd06629a5c55) #4 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           29:  unreachable 
check:13'0     ~~~~~~~~~~~~~
           30: } 
check:13'0     ~~
           31:  
check:13'0     ~
           32: ; core::panicking::panic_fmt 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           33: ; Function Attrs: cold noinline noreturn uwtable 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34: declare void @_RNvNtCsj5hlU51TwZT_4core9panicking9panic_fmt(ptr noundef nonnull, ptr noundef nonnull, ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #2 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35:  
check:13'0     ~
           36: attributes #0 = { uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           37: attributes #1 = { cold noreturn uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           38: attributes #2 = { cold noinline noreturn uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           39: attributes #3 = { noreturn } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           40: attributes #4 = { noinline noreturn } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           41:  
check:13'0     ~
           42: !llvm.module.flags = !{!0} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43: !llvm.ident = !{!1} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~
           44:  
check:13'0     ~
           45: !0 = !{i32 8, !"PIC Level", i32 2} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46: !1 = !{!"rustc version 1.94.0-nightly (a05bb8fcc 2025-12-25)"} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-20/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll" "/checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs:13:11: error: CHECK: expected string not found in input
// CHECK: br i1 %x, label %bb2, label %bb1, !prof ![[NUM:[0-9]+]]
          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll:10:26: note: scanning from here
define void @test_assert(i1 noundef zeroext %x) unnamed_addr #0 {
                         ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll:12:2: note: possible intended match here
 br i1 %x, label %bb2, label %bb1
 ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/intrinsics/likely_const_assert/likely_const_assert.ll
Check file: /checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: ; ModuleID = 'likely_const_assert.acc64894c2abf645-cgu.0' 
            2: source_filename = "likely_const_assert.acc64894c2abf645-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
            4: target triple = "aarch64-unknown-linux-gnu" 
            5:  
            6: @alloc_229a7d18cf7072cde189ca7a0a5192e3 = private unnamed_addr constant [63 x i8] c"/checkout/tests/codegen-llvm/intrinsics/likely_const_assert.rs\00", align 1 
            7: @alloc_a1222f3d3751f79e8911dd06629a5c55 = private unnamed_addr constant <{ ptr, [16 x i8] }> <{ ptr @alloc_229a7d18cf7072cde189ca7a0a5192e3, [16 x i8] c">\00\00\00\00\00\00\00\05\00\00\00\07\00\00\00" }>, align 8 
            8:  
            9: ; Function Attrs: uwtable 
           10: define void @test_assert(i1 noundef zeroext %x) unnamed_addr #0 { 
check:13'0                              X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
           11: start: 
check:13'0     ~~~~~~~
           12:  br i1 %x, label %bb2, label %bb1 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:13'1      ?                                 possible intended match
           13:  
check:13'0     ~
           14: bb1: ; preds = %start 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~
           15: ; call likely_const_assert::test_assert::do_panic::runtime 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           16:  tail call fastcc void @_RNvNvNvCsePFQ7nKV49d_19likely_const_assert11test_assert8do_panic7runtime() #3 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           17:  unreachable 
check:13'0     ~~~~~~~~~~~~~
           18:  
check:13'0     ~
           19: bb2: ; preds = %start 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~
           20:  ret void 
check:13'0     ~~~~~~~~~~
           21: } 
check:13'0     ~~
           22:  
check:13'0     ~
           23: ; likely_const_assert::test_assert::do_panic::runtime 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           24: ; Function Attrs: cold noreturn uwtable 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           25: define internal fastcc void @_RNvNvNvCsePFQ7nKV49d_19likely_const_assert11test_assert8do_panic7runtime() unnamed_addr #1 { 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           26: start: 
check:13'0     ~~~~~~~
           27: ; call core::panicking::panic_fmt 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           28:  tail call void @_RNvNtCsj5hlU51TwZT_4core9panicking9panic_fmt(ptr noundef nonnull inttoptr (i64 1 to ptr), ptr noundef nonnull inttoptr (i64 1 to ptr), ptr noalias noundef nonnull readonly align 8 dereferenceable(24) @alloc_a1222f3d3751f79e8911dd06629a5c55) #4 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           29:  unreachable 
check:13'0     ~~~~~~~~~~~~~
           30: } 
check:13'0     ~~
           31:  
check:13'0     ~
           32: ; core::panicking::panic_fmt 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           33: ; Function Attrs: cold noinline noreturn uwtable 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34: declare void @_RNvNtCsj5hlU51TwZT_4core9panicking9panic_fmt(ptr noundef nonnull, ptr noundef nonnull, ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #2 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35:  
check:13'0     ~
           36: attributes #0 = { uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           37: attributes #1 = { cold noreturn uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           38: attributes #2 = { cold noinline noreturn uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           39: attributes #3 = { noreturn } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           40: attributes #4 = { noinline noreturn } 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           41:  
check:13'0     ~
           42: !llvm.module.flags = !{!0} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43: !llvm.ident = !{!1} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~
           44:  
check:13'0     ~
           45: !0 = !{i32 8, !"PIC Level", i32 2} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46: !1 = !{!"rustc version 1.94.0-nightly (a05bb8fcc 2025-12-25)"} 
check:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
------------------------------------------

---- [codegen] tests/codegen-llvm/intrinsics/likely_const_assert.rs stdout end ----

```

</details>

--- a/library/core/src/intrinsics/mod.rs
+++ b/library/core/src/intrinsics/mod.rs
@@ -2402,8 +2402,7 @@ where
 /// The `@capture` block declares which surrounding variables / expressions can be
 /// used inside the `if const`.
 /// Note that the two arms of this `if` really each become their own function, which is why the
-/// macro supports setting attributes for those functions. The runtime function is always
-/// marked as `#[inline]`.
+/// macro supports setting attributes for those functions. Both functions are marked as `#[inline]`.
 ///
 /// See [`const_eval_select()`] for the rules and requirements around that intrinsic.
 pub(crate) macro const_eval_select {
@@ -2413,35 +2412,14 @@ pub(crate) macro const_eval_select {
             $(#[$compiletime_attr:meta])* $compiletime:block
         else
             $(#[$runtime_attr:meta])* $runtime:block
-    ) => {
-        // Use the `noinline` arm, after adding explicit `inline` attributes
-        $crate::intrinsics::const_eval_select!(
-            @capture$([$($binders)*])? { $($arg : $ty = $val),* } $(-> $ret)? :
-            #[noinline]
-            if const
-                #[inline] // prevent codegen on this function
-                $(#[$compiletime_attr])*
-                $compiletime
-            else
-                #[inline] // avoid the overhead of an extra fn call
-                $(#[$runtime_attr])*
-                $runtime
-        )
-    },
-    // With a leading #[noinline], we don't add inline attributes
-    (
-        @capture$([$($binders:tt)*])? { $($arg:ident : $ty:ty = $val:expr),* $(,)? } $( -> $ret:ty )? :
-        #[noinline]
-        if const
-            $(#[$compiletime_attr:meta])* $compiletime:block
-        else
-            $(#[$runtime_attr:meta])* $runtime:block
     ) => {{
+        #[inline]
         $(#[$runtime_attr])*
         fn runtime$(<$($binders)*>)?($($arg: $ty),*) $( -> $ret )? {
             $runtime
         }
 
+        #[inline]
         $(#[$compiletime_attr])*
         const fn compiletime$(<$($binders)*>)?($($arg: $ty),*) $( -> $ret )? {
             // Don't warn if one of the arguments is unused.

--- a/library/core/src/panic.rs
+++ b/library/core/src/panic.rs
@@ -166,10 +166,9 @@ pub macro const_panic {
         const fn do_panic($($arg: $ty),*) -> ! {
             $crate::intrinsics::const_eval_select!(
                 @capture { $($arg: $ty = $arg),* } -> !:
-                #[noinline]
-                if const #[track_caller] #[inline] { // Inline this, to prevent codegen
+                if const #[track_caller] {
                     $crate::panic!($const_msg)
-                } else #[track_caller] { // Do not inline this, it makes perf worse
+                } else #[track_caller] {
                     $crate::panic!($runtime_msg)
                 }
             )
@@ -195,7 +194,7 @@ pub macro const_panic {
 #[doc(hidden)]
 pub macro const_assert {
     ($condition: expr, $const_msg:literal, $runtime_msg:literal, $($arg:tt)*) => {{
-        if !$crate::intrinsics::likely($condition) {
+        if !($condition) {
             $crate::panic::const_panic!($const_msg, $runtime_msg, $($arg)*)
         }
     }}

--- a/tests/codegen-llvm/intrinsics/likely_assert.rs
+++ b/tests/codegen-llvm/intrinsics/likely_assert.rs
@@ -1,12 +1,13 @@
 //@ compile-flags: -Copt-level=3
+#![feature(panic_internals, const_eval_select, rustc_allow_const_fn_unstable, core_intrinsics)]
 #![crate_type = "lib"]
+
+// check that assert! and const_assert! emit branch weights
 
 #[no_mangle]
 pub fn test_assert(x: bool) {
     assert!(x);
 }
-
-// check that assert! emits branch weights
 
 // CHECK-LABEL: @test_assert(
 // CHECK: br i1 %x, label %bb2, label %bb1, !prof ![[NUM:[0-9]+]]
@@ -14,4 +15,17 @@ pub fn test_assert(x: bool) {
 // CHECK: panic
 // CHECK: bb2:
 // CHECK: ret void
+
+#[no_mangle]
+pub fn test_const_assert(x: bool) {
+    core::panic::const_assert!(x, "", "",);
+}
+
+// CHECK-LABEL: @test_const_assert(
+// CHECK: br i1 %x, label %bb2, label %bb1, !prof ![[NUM:[0-9]+]]
+// CHECK: bb1:
+// CHECK: panic
+// CHECK: bb2:
+// CHECK: ret void
+
 // CHECK: ![[NUM]] = !{!"branch_weights", {{(!"expected", )?}}i32 2000, i32 1}

